### PR TITLE
BUG: Series.take needs to validate axis

### LINKED ISSUE #51022: BUG: Series.take needs to validate axis
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
pd.Series([1, 2, 3]).take([1], axis="foo")
```

### Issue Description

Nonsense "axis" fails to raise

### Expected Behavior

ValueError

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by aneesh98**
- [X] closes #51022 (Replace xxxx with the GitHub issue number)
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -890,7 +890,8 @@ def axes(self) -> list[Index]:
     @Appender(NDFrame.take.__doc__)
     def take(self, indices, axis: Axis = 0, **kwargs) -> Series:
         nv.validate_take((), kwargs)
-
+        if axis not in [0, 1, "index", "columns", None]:
```
Comment (jbrockmendel) [inline pandas/core/series.py:893]: try just `self._get_axis_number(axis)`

Comment (aneesh98) [reply pandas/core/series.py]: Hi @jbrockmendel  Can you please check now?

File: pandas/core/series.py [hunk 2]
```diff
@@ -891,6 +891,9 @@ def axes(self) -> list[Index]:
     def take(self, indices, axis: Axis = 0, **kwargs) -> Series:
         nv.validate_take((), kwargs)
 
+        # GH # 51022
```
Comment (jbrockmendel) [inline pandas/core/series.py:894]: nitpick can you make this just `# GH#51022`

File: pandas/tests/series/indexing/test_take.py [hunk 3]
```diff
@@ -31,3 +31,9 @@ def test_take_categorical():
         pd.Categorical(["b", "b", "a"], categories=["a", "b", "c"]), index=[1, 1, 0]
     )
     tm.assert_series_equal(result, expected)
+
+def test_take_incorrect_axis_name():
```
Comment (jbrockmendel) [inline pandas/tests/series/indexing/test_take.py:35]: you're gonna need an extra newline before this

Comment (aneesh98) [reply pandas/tests/series/indexing/test_take.py:36]: Done

**Comment by jbrockmendel** [comment]
the linter is complaining about something.  take a look

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**